server/prom/query.py

```diff
@@ -72,10 +72,11 @@ def query(self):
             df = pd.DataFrame(items) 
             df.columns = df.columns.str.replace("curr_", "")
             df.columns = df.columns.str.replace("node_", "")
-            df[query_metric] = df['value']
-            for col in df.columns:
-                df[col] = df[col].transform(transform_float)
-            df.drop(columns=['value'], inplace=True)
+            if len(df) > 0:
```

Comment:
Is it possible for df to be nan or is it always an empty string? Might need to consider pandas isnull function.

Reply:
df will be always initialized by `pd.DataFrame(items) `. with no data, items = []. and df will just has no data